### ISSUE TITLE
有没有大神可以写一个FastText输出预测类的，谢谢^_^

### ISSUE THREAD

**Issue by dkdaoren** [649453932/Chinese-Text-Classification-Pytorch#90]
希望有大神可以写个FastText输出预测类的给代码也行，谢谢

**Comment by T00ler** [comment]
> 希望有大神可以写个FastText输出预测类的给代码也行，谢谢

hxd，FastText的预测写好了吗，求一份

**Comment by JieJayCao** [comment]
> hxd，FastText的预测写好了吗，求一份

hxd, 有fastText 的预测吗，或者其他几个模型的预测也行，求求

**Comment by cingtiye** [comment]
> 希望有大神可以写个FastText输出预测类的给代码也行，谢谢

https://github.com/cingtiye/Chinese_question_sentence_judgment/blob/master/FastText/evalute.py

**Comment by cingtiye** [comment]
https://github.com/cingtiye/Chinese_question_sentence_judgment/blob/master/FastText/evalute.py

**Comment by xiaomingtx6** [comment]
# -*- coding: utf-8 -*-
import torch
import pickle as pkl
from config import Config
from model import Model

key = {
    0: 'no',
    1: 'yes',
}
UNK, PAD = '<UNK>', '<PAD>'
config = Config("/home/adt01/Chinese-Text-Classification-Pytorch-master/THUCNews/", "random")
tokenizer = lambda x: [y for y in x]
model = Model(config).to("cpu")

model.load_state_dict(torch.load("/home/adt01/Chinese-Text-Classification-Pytorch-master/THUCNews/saved_dict/FastText.ckpt", map_location='cpu'))


def build_predict_text(text, pad_size=32):
    text = text.split("\t")
    vocab = pkl.load(open("./THUCNews/data/vocab.pkl", 'rb'))
    config.n_vocab = len(vocab)

    def biGramHash(sequence, t, buckets):
        t1 = sequence[t - 1] if t - 1 >= 0 else 0
        return (t1 * 14918087) % buckets

    def triGramHash(sequence, t, buckets):
        t1 = sequence[t - 1] if t - 1 >= 0 else 0
        t2 = sequence[t - 2] if t - 2 >= 0 else 0
        return (t2 * 14918087 * 18408749 + t1 * 14918087) % buckets

    label = 0
    words_line = []
    token = tokenizer(text)
    if pad_size:
        if len(token) < pad_size:
            token.extend([PAD] * (pad_size - len(token)))
        else:
            token = token[:pad_size]
    # word to id
    for word in token:
        words_line.append(vocab.get(word, vocab.get(UNK)))

    # fasttext ngram
    buckets = config.n_gram_vocab
    bigram = []
    trigram = []
    # ------ngram------
    for i in range(pad_size):
        bigram.append(biGramHash(words_line, i, buckets))
        trigram.append(triGramHash(words_line, i, buckets))
    # x, seq_len, mask, seg_ids
    return words_line, int(label), bigram, trigram


def predict(text):
    """
    单个文本预测
    :param text:
    :return:
    """

    data = build_predict_text(text)
    with torch.no_grad():
        outputs = model(data)
        num = torch.argmax(outputs.data, 1).cpu().numpy()
    return key[int(num)], torch.softmax(outputs, dim=1)


if __name__ == '__main__':
    print(predict("我喜欢电竞游戏"))
